### PR TITLE
#30 VPC Public IP

### LINKED ISSUE #30: VServer > Public IP
#### Resource
- [x] resource_ncloud_public_ip (Modify)
- [ ] resource_ncloud_public_ip_association (New)

#### Data source
- [x] data_source_ncloud_public_ip (Modify)

#### API
- getPublicIpInstanceList
- getPublicIpInstanceDetail
- createPublicIpInstance
- deletePublicIpInstance
- getPublicIpTargetServerInstanceList
- associatePublicIpWithServerInstance
- disassociatePublicIpFromServerInstance

### PR DESCRIPTION
**Pull request by sangkyu-kim**
resolve #30 

### Features
- Support VPC 

#### Resource
- [x] resource_ncloud_public_ip (Modify)
##### Add Features
- update public ip if `server_instance_no` has changed

#### Data source
- [x] data_source_ncloud_public_ip (Modify)
##### Add
- filter

##### Deprecated
- search_filter_name
- search_filter_value
- region
- sorted_by
- sorting_order

### CODE REVIEW

File: ncloud/data_source_ncloud_public_ip.go [hunk 1]
```diff
@@ -126,21 +133,42 @@ func dataSourceNcloudPublicIp() *schema.Resource {
...
 
-	regionNo, err := parseRegionNoParameter(client, d)
 	if err != nil {
 		return err
 	}
-	zoneNo, err := parseZoneNoParameter(client, d)
-	if err != nil {
+
+	if f, ok := d.GetOk("filter"); ok {
+		resources = ApplyFilters(f.(*schema.Set), resources, dataSourceNcloudMemberServerImage().Schema)
```
Comment (YuSungDuk) [inline ncloud/data_source_ncloud_public_ip.go]: dataSourceNcloudMemberServerImage??  dataSourceNcloudPublicIp?? resourceNcloudPublicIpInstance??

File: ncloud/resource_ncloud_public_ip_test.go [hunk 2]
```diff
@@ -53,11 +51,40 @@ func TestAccResourceNcloudPublicIpInstanceBasic(t *testing.T) {
 	})
 }
 
-func testAccCheckPublicIpInstanceExists(n string, i *server.PublicIpInstance) resource.TestCheckFunc {
-	return testAccCheckPublicIpInstanceExistsWithProvider(n, i, func() *schema.Provider { return testAccProvider })
+func TestAccResourceNcloudPublicIpInstance_update(t *testing.T) {
+	instance := map[string]interface{}{}
+	serverNameFoo := fmt.Sprintf("test-public-ip-basic-%s", acctest.RandString(5))
+	serverNameBar := fmt.Sprintf("test-public-ip-basic-%s", acctest.RandString(5))
```
Comment (YuSungDuk) [inline ncloud/resource_ncloud_public_ip_test.go]: 낮은 확률로 동일 이름이 될 가능성이 있을듯 합니다.

File: ncloud/resource_ncloud_public_ip_test.go [hunk 3]
```diff
@@ -53,11 +51,40 @@ func TestAccResourceNcloudPublicIpInstanceBasic(t *testing.T) {
...
+	resourceName := "ncloud_public_ip.public_ip"
+
+	log.Print(testAccPublicIpInstanceConfigServer(serverNameFoo, serverNameBar, "${ncloud_server.foo.id}"))
```
Comment (YuSungDuk) [inline ncloud/resource_ncloud_public_ip_test.go]: 로깅 남겨두어야 하나요?

**Review by YuSungDuk — CHANGES_REQUESTED** [review]

File: ncloud/data_source_ncloud_public_ip.go (hunk 1 above)
Comment (sangkyu-kim) [reply ncloud/data_source_ncloud_public_ip.go]: `dataSourceNcloudPublicIp` 로 변경 하였습니다.

**Review by YuSungDuk — APPROVED** [review]
👍 Good night~

**Pull request merged**